From 40d3d03d28c13a34547db38e40b74bdd4195848e Mon Sep 17 00:00:00 2001
From: Michael Schroeder <mls@suse.de>
Date: Wed, 25 Oct 2017 14:23:02 +0200
Subject: [PATCH 3/3] Fix droporphaned when there is no installed repo

---
 src/solver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/solver.c b/src/solver.c
index e90f8a4..b9d91d2 100644
--- a/src/solver.c
+++ b/src/solver.c
@@ -3588,7 +3588,7 @@ solver_solve(Solver *solv, Queue *job)
 		}
 	      break;
 	    case SOLVER_DROP_ORPHANED:
-	      if (select == SOLVER_SOLVABLE_ALL || (select == SOLVER_SOLVABLE_REPO && what == installed->repoid))
+	      if (select == SOLVER_SOLVABLE_ALL || (select == SOLVER_SOLVABLE_REPO && installed && what == installed->repoid))
 		solv->droporphanedmap_all = 1;
 	      FOR_JOB_SELECT(p, pp, select, what)
 		{
-- 
2.13.5

