From e9292a3a41d357d88938e62a4fabfc1d54f50741 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Sun, 17 Sep 2017 06:11:21 +0000
Subject: [PATCH] ext/repo_rpmdb: Use the correct BDB flags for RPM5

---
 ext/repo_rpmdb.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/ext/repo_rpmdb.c b/ext/repo_rpmdb.c
index 92aef31..ec0f032 100644
--- a/ext/repo_rpmdb.c
+++ b/ext/repo_rpmdb.c
@@ -1172,7 +1172,11 @@ opendbenv(struct rpmdbstate *state)
   if (db_env_create(&dbenv, 0))
     return pool_error(state->pool, 0, "db_env_create: %s", strerror(errno));
 #if defined(FEDORA) && (DB_VERSION_MAJOR >= 5 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 5))
+#ifdef RPM5
+  dbenv->set_thread_count(dbenv, 64);
+#else
   dbenv->set_thread_count(dbenv, 8);
+#endif
 #endif
   snprintf(dbpath, PATH_MAX, "%s/var/lib/rpm", rootdir ? rootdir : "");
   if (access(dbpath, W_OK) == -1)
@@ -1187,7 +1191,11 @@ opendbenv(struct rpmdbstate *state)
     {
 #ifdef FEDORA
       int serialize_fd = serialize_dbenv_ops(state);
+#ifdef RPM5
+      r = dbenv->open(dbenv, dbpath, DB_CREATE|DB_THREAD|DB_INIT_LOCK|DB_INIT_LOG|DB_INIT_MPOOL|DB_INIT_TXN, 0644);
+#else
       r = dbenv->open(dbenv, dbpath, DB_CREATE|DB_INIT_CDB|DB_INIT_MPOOL, 0644);
+#endif
       if (serialize_fd >= 0)
 	close(serialize_fd);
 #else
-- 
2.14.1

