diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd1426b..2ba0a52 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,7 @@ OPTION (ENABLE_TCL "Build the Tcl bindings?" OFF)
 
 OPTION (USE_VENDORDIRS "Install the bindings in vendor directories?" OFF)
 
+OPTION (ENABLE_RPM5 "Build with rpm 5.x support?" OFF)
 OPTION (ENABLE_RPMDB "Build with rpm database support?" OFF)
 OPTION (ENABLE_PUBKEY "Build with pubkey support?" OFF)
 OPTION (ENABLE_RPMDB_BYRPMHEADER "Build with rpmdb Header support?" OFF)
@@ -160,7 +161,7 @@ IF (ENABLE_BZIP2_COMPRESSION)
 FIND_PACKAGE (BZip2 REQUIRED)
 ENDIF (ENABLE_BZIP2_COMPRESSION)
 
-IF (RPM5)
+IF (ENABLE_RPM5)
 MESSAGE (STATUS "Enabling RPM 5 support")
 ADD_DEFINITIONS (-DRPM5)
 SET (ENABLE_RPMDB ON)
@@ -168,7 +169,7 @@ SET (ENABLE_RPMMD ON)
 FIND_PACKAGE (PkgConfig REQUIRED)
 PKG_CHECK_MODULES (RPM REQUIRED rpm)
 INCLUDE_DIRECTORIES (${RPM_INCLUDE_DIRS})
-ENDIF (RPM5)
+ENDIF (ENABLE_RPM5)
 
 IF (MULTI_SEMANTICS)
 MESSAGE (STATUS "Enabling multi dist support")
@@ -188,12 +189,12 @@ IF (ENABLE_RPMDB)
     SET(RPMDB_LIBRARY ${RPMIO_LIBRARY} ${RPMDB_LIBRARY})
   ENDIF (RPMIO_LIBRARY)
 
-  IF (RPM5)
+  IF (ENABLE_RPM5)
     FIND_LIBRARY (RPMMISC_LIBRARY NAMES rpmmisc)
     IF (RPMMISC_LIBRARY)
       SET (RPMDB_LIBRARY ${RPMMISC_LIBRARY} ${RPMDB_LIBRARY})
     ENDIF (RPMMISC_LIBRARY)
-  ENDIF (RPM5)
+  ENDIF (ENABLE_RPM5)
 
   # check if rpm contains a bundled berkeley db
   CHECK_INCLUDE_FILE(rpm/db.h HAVE_RPM_DB_H)
diff --git a/bindings/python/CMakeLists.txt b/bindings/python/CMakeLists.txt
index 73f2dca..685cb95 100644
--- a/bindings/python/CMakeLists.txt
+++ b/bindings/python/CMakeLists.txt
@@ -26,7 +26,7 @@ INCLUDE_DIRECTORIES (${PYTHON_INCLUDE_PATH})
 
 ADD_LIBRARY (bindings_python MODULE solv_python.c)
 SET_TARGET_PROPERTIES (bindings_python PROPERTIES PREFIX "" OUTPUT_NAME "_solv")
-TARGET_LINK_LIBRARIES (bindings_python libsolvext libsolv ${SYSTEM_LIBRARIES})
+TARGET_LINK_LIBRARIES (bindings_python libsolvext libsolv ${SYSTEM_LIBRARIES} -lpython2.7)
 
 INSTALL (TARGETS bindings_python LIBRARY DESTINATION ${PYTHON_INSTALL_DIR})
 INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/solv.py DESTINATION ${PYTHON_INSTALL_DIR})
